plutus-ledger-api/src/Plutus/V1/Ledger/Tx.hs

```diff
@@ -265,6 +281,20 @@ txId tx = TxId $ BA.convert h' where
     h' :: Digest SHA256
     h' = hash h
 
+-- | A tag indicating the type of script that we are pointing to.
+-- NOTE: Cert/Reward are not supported right now.
+data ScriptTag = Spend | Mint | Cert | Reward
+    deriving stock (Show, Eq, Ord, Generic)
+    deriving anyclass (Serialise, ToJSON, FromJSON, NFData)
+
+-- | A redeemer pointer is a pair of a script type tag t and an index i, picking out the ith
+-- script of type t in the transaction.
+data RedeemerPtr = RedeemerPtr ScriptTag Integer
+    deriving stock (Show, Eq, Ord, Generic)
+    deriving anyclass (Serialise, ToJSON, FromJSON, ToJSONKey, FromJSONKey, NFData)
+
+type Redeemers = Map RedeemerPtr Redeemer
```

Comment:
Doing it this way is fine, but it's closer to the `cardano-ledger-specs` version of `Tx` rather than the `cardano-api` one. In `cardano-api` it looks like they have separate maps for each of the different kinds of things, e.g. the data type for withdrawals includes the witnesses which includes the redeemers: https://github.com/input-output-hk/cardano-node/blob/3d6a249386ae86be63ad0a81006d35d22f2e3d4e/cardano-api/src/Cardano/Api/TxBody.hs#L974

So I don't know if we want to keep this representation for now or whether we should try and do something closer to what `cardano-api` does?

Reply:
Since we don't have `BuildTxWith` I'm not so sure that will be much of an improvement.